### ISSUE TITLE
push to master restriction

### ISSUE THREAD

**Issue by Symous** [gitblit/gitblit#1311]
Awesome lib! 
Is there any way that we can configure a branch to restricting push to master branch, such as "protected branch" features? I just want to allow a few users to push or merge to master branch.

**Comment by fzs** [comment]
No, unfortunately branch protection is not yet implemented, but certainly high on the Wanted list.

**Comment by Symous** [comment]
@fzs thanks for the reply. I am looking forward to this feature. :)

**Comment by chirontt** [comment]
I think what @Symous needs is already available in the form of a pre-receive hook script.

The existing hook script protect-refs.groovy (in data/groovy folder) does similar thing for preventing push to master. It just need to be modified slightly to do what @Symous wants.

**Comment by Symous** [comment]
@chirontt Thanks friend, I have check protect-refs.groovy and seems like that it can truly achieve my requirement(not test yet). It will be much better if we can configure that on Gitblit page.

**Comment by chirontt** [comment]
Another option is to use the Gitblit plugin **tbacl** (Team-Branch ACLs) available at the [Gitblit Registry](http://gitblit.github.io/gitblit-registry/). The plugin's source code is [here](https://github.com/gitblit/gitblit-tbacl-plugin), which was developed by James.
